hera_cal/frf.py

```diff
@@ -10,22 +10,165 @@
     HAVE_UVTOOLS = False
 
 from . import utils
-
+from scipy.interpolate import interp1d
 from .datacontainer import DataContainer
 from .vis_clean import VisClean
-from pyuvdata import UVData, UVFlag
+from pyuvdata import UVData, UVFlag, UVBeam
 import argparse
 from . import io
 from . import vis_clean
 import warnings
+from astropy.coordinates import SkyCoord, EarthLocation, AltAz, ITRS
+from astropy import units
 import astropy.constants as const
-
+from astropy_healpix import HEALPix
+from astropy.time import Time
+from pyuvdata import utils as uvutils
+from . import utils
+import copy
+from .utils import echo
+import datetime
+import astropy.constants as const
+from . import redcal
 
 SPEED_OF_LIGHT = const.c.si.value
-SDAY_KSEC = 86163.93 / 1000.
+SDAY_KSEC = units.sday.to("ks")
+
+
+def _get_key_reds(antpos, keys):
+    """
+    Private function to get redundant groups in list of antpairpol keys.
+
+    Parameters
+    ----------
+    antpos: dict
+        dictionary with integer keys mapping to length 3 numpy array ENU antenna positions.
+    keys:
+        list of antpairpols to split into redundant groups.
+
+    Returns
+    -------
+    reds: list of lists of tuples
+        list of lists where each sublist is a redundant group
+        and each element of each redundant group is a redundant antpairpol
+        in keys.
+    """
+    # get redundancies (will only compute fr-profile once for each red group).
+    unique_pols = set()
+    for bl in keys:
+        if bl[-1] not in unique_pols:
+            unique_pols.add(bl[-1])
+    reds = redcal.get_reds(antpos, pols=list(unique_pols), include_autos=True)
+    reds = redcal.filter_reds(reds, bls=keys)
+    return reds
+
+
+def select_tophat_frates(uvd, blvecs, case, keys=None, frate_standoff=0.0, frate_width_multiplier=1.0, min_frate_half_width=0.025,
```

Comment:
can we make this

```
select_tophat_frates(case, keys=None, blvecs=None, uvd=None, ...):
    ...
    elif case == 'max_frate_coeffs':
        keys = blvecs.keys()

```
i.e. case should be first argument, and uvd should be optional b/c its not strictly needed for the max_frate_coeeffs case

Reply:
done.